IRSS-UBC/3pg
# Fix allowable opvars

What this PR does:
 - add output vars to dataInput
 - fix bugs with year planted and frost days
 - edit tests to run with newer version
 - change the way the logger is called (because it was causing weeeeeird issues)

how to test:
 - run unit tests
 - run through different scenarios on the executable with different output variables, and ensure the model runs when it has the correct opvars but not when it doesn't. Also, ensure logging is working correctly.

(probably a good idea to squash this one as it was started from the shared pointer branch before that one got merged, and has a couple commits to just merge and stay up to date with main.)
